src/instruments/src/PFD/FMA.tsx

```diff
@@ -1,554 +1,1021 @@
-        text = 'G/S*';
-        break;
-    // case 4:
-    //     text = 'F-G/S*';
-    //     break;
-    case VerticalMode.SRS:
-    case VerticalMode.SRS_GA:
-        text = 'SRS';
-        break;
-    case VerticalMode.TCAS:
-        text = 'TCAS';
-        break;
-    // case 9:
-    //     text = 'FINAL';
-    //     break;
-    case VerticalMode.DES:
-        text = 'DES';
-        break;
-    case VerticalMode.OP_DES:
-        if (getSimVar('L:A32NX_FMA_EXPEDITE_MODE', 'bool')) {
-            text = 'EXP DES';
-        } else {
-            text = 'OP DES';
+class B1Cell extends ShowForSecondsComponent<CellProps> {
+    private boxClassSub = Subject.create('');
+
+    private boxPathStringSub = Subject.create('');
+
+    private activeVerticalModeSub = Subject.create(0);
+
+    private inProtectionClassSub = Subject.create('Cyan');
+
+    private speedProtectionPathRef = FSComponent.createRef<SVGPathElement>();
+
+    private inModeReversionPathRef = FSComponent.createRef<SVGPathElement>();
+
+    private fmaTextRef = FSComponent.createRef<SVGTextElement>();
+
+    private selectedVS = 0;
+
+    private inSpeedProtection = 0;
+
+    private expediteMode = false;
+
+    private FPA = 0;
+
+    constructor(props: CellProps) {
+        super(props, 10);
+    }
+
+    private getText(): boolean {
+        let text: string;
+        let additionalText: string = '';
+
+        this.isShown = true;
+        switch (this.activeVerticalModeSub.get()) {
+        case 31:
+            text = 'G/S';
+            break;
+        // case 2:
+        //     text = 'F-G/S';
+        //     break;
+        case 30:
+            text = 'G/S*';
+            break;
+        // case 4:
+        //     text = 'F-G/S*';
+        //     break;
+        case 40:
+        case 41:
+            text = 'SRS';
+            break;
+        case 50:
+            text = 'TCAS';
+            break;
+        // case 9:
+        //     text = 'FINAL';
+        //     break;
+        case 23:
+            text = 'DES';
+            break;
+        case 13:
+            if (this.expediteMode) {
+                text = 'EXP DES';
+            } else {
+                text = 'OP DES';
+            }
+            break;
+        case 22:
+            text = 'CLB';
+            break;
+        case 12:
+            if (this.expediteMode) {
+                text = 'EXP CLB';
+            } else {
+                text = 'OP CLB';
+            }
+            break;
+        case 10:
+            if (SimVar.GetSimVarValue('L:A32NX_FMA_CRUISE_ALT_MODE', 'Bool')) {
+                text = 'ALT CRZ';
+            } else {
+                text = 'ALT';
+            }
+            break;
+        case 11:
+            text = 'ALT*';
+            break;
+        case 21:
+            text = 'ALT CST*';
+            break;
+        case 20:
+            text = 'ALT CST';
+            break;
+        // case 18:
+        //     text = 'ALT CRZ';
+        //     break;
```

Comment:
Why is this commented? I believe we support this mode.

Reply:
This was copied straight from the existing FMA code, but yes we can activate it. Shouldn't cause any harm.

Reply:
This is implemented a few lines up, within the "normal" ALT mode, by checking the lvar A32NX_FMA_CRUISE_ALT_MODE